flink-connector-influxdb2/src/test/java/org/apache/flink/streaming/connectors/influxdb/InfluxDBSourceIntegrationTestCase.java

```diff
@@ -67,59 +68,76 @@
 
     private final StreamExecutionEnvironment env =
             StreamExecutionEnvironment.getExecutionEnvironment().setParallelism(1);
-    private final InfluxDBSource<Long> influxDBSource =
-            InfluxDBSource.<Long>builder()
-                    .setPort(PORT)
-                    .setDeserializer(new InfluxDBTestDeserializer())
-                    .build();
 
-    @BeforeAll
-    static void setUp() {
+    @BeforeEach
+    void init() {
         CollectSink.VALUES.clear();
+        try (final ServerSocket serverSocket = new ServerSocket(0)) {
+            this.port = serverSocket.getLocalPort();
+            this.log.info("Using port {} for the HTTP server", this.port);
+        } catch (final IOException ioException) {
+            this.log.error("Could not open open port {}", ioException.getMessage());
+        }
     }
 
     /**
      * Test the following topology.
      *
      * <pre>
-     *     1,2,3                +1              2,3,4
-     *     (source1/1) -----> (map1/1) -----> (sink1/1)
+     *     test longValue=1i 1     +1            2
+     *     test longValue=2i 1     +1            3
+     *     (source) ------------> (map) -----> (sink)
      * </pre>
      */
     @Test
     void testIncrementPipeline() throws Exception {
+        final InfluxDBSource<Long> influxDBSource =
+                InfluxDBSource.<Long>builder()
+                        .setPort(this.port)
+                        .setDeserializer(new InfluxDBTestDeserializer())
+                        .build();
+
         this.env
-                .fromSource(this.influxDBSource, WatermarkStrategy.noWatermarks(), "InfluxDBSource")
+                .fromSource(influxDBSource, WatermarkStrategy.noWatermarks(), "InfluxDBSource")
                 .map(new IncrementMapFunction())
                 .addSink(new CollectSink());
 
         final JobClient jobClient = this.env.executeAsync();
-        assertTrue(checkHealthCheckAvailable());
+        assertTrue(this.checkHealthCheckAvailable());
 
-        final int writeResponseCode = writeToInfluxDB("test longValue=1i 1");
+        final int writeResponseCode =
+                this.writeToInfluxDB("test longValue=1i 1\ntest longValue=2i 2");
 
         assertEquals(writeResponseCode, HttpURLConnection.HTTP_NO_CONTENT);
 
         jobClient.cancel();
 
         final Collection<Long> results = new ArrayList<>();
         results.add(2L);
+        results.add(3L);
+        Thread.sleep(500);
```

Comment:
Not good, hardcode a thread sleep here is a mistake by design. Should think about something else

Reply:
```suggestion
```